flowmachine/flowmachine/features/location/flows.py

```diff
@@ -62,15 +62,56 @@ def _geo_augmented_query(self):
         self.spatial_unit.verify_criterion("has_geography")
 
         loc_cols = self.spatial_unit.location_id_columns
+        if hasattr(self, "out_label_columns"):
+            lab_cols = self.out_label_columns
+        else:
+            lab_cols = []
         loc_cols_string = ",".join(loc_cols)
-        loc_cols_from_string = ",".join([f"{col}_from" for col in loc_cols])
-        loc_cols_to_string = ",".join([f"{col}_to" for col in loc_cols])
-        loc_cols_from_aliased_string = ",".join(
-            [f"{col}_from AS {col}" for col in loc_cols]
-        )
-        loc_cols_to_aliased_string = ",".join(
-            [f"{col}_to AS {col}" for col in loc_cols]
-        )
+
+        def _build_json_agg_clause(loc_cols, direction, label_cols=[]):
```

Comment:
Can this be split out as a separate method, instead of a function defined within `_geo_augmented_query`?

Reply:
Done + test added to keep covbot happy